storage/pipeline/input.go

```diff
@@ -28,8 +29,13 @@ import (
 
 func (m *Sealing) handleWaitDeals(ctx statemachine.Context, sector SectorInfo) error {
 	var used abi.UnpaddedPieceSize
+	var lastDealEnd abi.ChainEpoch
 	for _, piece := range sector.Pieces {
 		used += piece.Piece.Size.Unpadded()
+
+		if piece.DealInfo.DealProposal.EndEpoch > lastDealEnd {
```

Comment:
Because piece.DealInfo is a pointer, should check if it is nil first?

Reply:
Thank you for the review!